nav2_map_server/README.md

```diff
@@ -109,36 +249,237 @@ and could be invoked by following command:
+
+#### Composible nodes
+
+There is also possibility of having multiple map server/saver nodes of different types in a single process, where each node parameters would be separated by its name like presented below:
+
+- Map server
+```yaml
+# For composite MapServer launch
+# examples/multi_map_server_node/multi_composite_node_params.yaml
+map_server_2d_n1:
+    ros__parameters:
+        yaml_filename: "./occupancy_grid_map.yaml"
+map_server_2d_n2:
+    ros__parameters:
+        yaml_filename: "./occupancy_grid_map.yaml"
+map_server_3d_n3:
+    ros__parameters:
+        yaml_filename: "./point_cloud_map.yaml"
+map_server_3d_n4:
+    ros__parameters:
+        yaml_filename: "./point_cloud_map.yaml"
+```
+
+- Map server
+```yaml
+# For composite MapSaver launch
+# examples/multi_map_saver_node/multi_composite_node_params.yaml
+map_saver_server_2d_1:
+    ros__parameters:
+        save_map_timeout: 2.0
+        free_thresh_default: 0.25
+        occupancy_thresh_default: 0.65
+map_saver_server_2d_2:
+    ros__parameters:
+        save_map_timeout: 2.0
+        free_thresh_default: 0.25
+        occupancy_thresh_default: 0.5
+map_saver_server_3d_3:
+    ros__parameters:
+        save_map_timeout: 5.0
+map_saver_server_3d_4:
+    ros__parameters:
+        save_map_timeout: 4.0
+```
+
+Then, one can invoke the process via launch file, with the YAML-file that contains the parameters for both nodes:
+
+```shell
+# For Map Server
+# Files are placed in the
+# `nav2_map_server/examples/multi_map_saver_node` directory
+$ ros2 launch multi_composite_map_saver.launch.py params_file:="multi_composite_node_params.yaml"
+# For Map Saver
+# Files are placed in the
+# `nav2_map_server/examples/multi_map_server_node` directory
+$ ros2 launch multi_composite_map_server.launch.py params_file:="multi_composite_node_params.yaml"
+```
+
+## Supported Map Types
 
 - Occupancy grid (nav_msgs/msg/OccupancyGrid)
+- PointCloud PointCloud2 (sensor_msgs/msg/PointCloud2)
+
+For now, there are only two main map types are supported one for `2D`(OccupancyGrid) and other for `3D`(PointCloud), still one can implement their own with full freedom to hold all the aspects of the new map type. This is simple with Nav2 Map Server, just define the *template specialization* of the core(bare minimum) implementation for Map Server`nav2_map_server/map_server_core.<hpp/cpp>`, and Map Saver`nav2_map_server/map_saver_core.<hpp/cpp>`. The new map type can use either a preimplemented `MapIO 2D` and `MapIO 3D`, or a custom I/O library to handle map files, and message conversions.
+
+**Note:** In order to get *templates* working properly, the new *template specialization* should be linked with the core implentation of Map Server and Saver. In the present implementation this is done through a top-level header for Map Server`nav2_map_server/map_server.hpp` and Map Saver`nav2_map_server/map_saver.hpp`.
```

Comment:
This is the detail of implementation: both `map_server.hpp` and `map_saver.hpp` headers are collective headers  where new specifications should be added. There are many other items like this are remain untold. So, I'd rather suggest to remove this notice and leave it for the tutorial.